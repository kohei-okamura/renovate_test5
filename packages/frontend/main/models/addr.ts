/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Prefecture } from '@zinger/enums/lib/prefecture'

/**
 * 住所.
 */
export type Addr = Readonly<{
  /** 郵便番号 */
  postcode: string

  /** 都道府県 */
  prefecture: Prefecture

  /** 市区町村 */
  city: string

  /** 町名・番地 */
  street: string

  /** 建物名など */
  apartment: string
}>
