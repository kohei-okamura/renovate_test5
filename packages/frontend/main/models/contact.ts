/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { ContactRelationship } from '@zinger/enums/lib/contact-relationship'

/**
 * 連絡先電話番号.
 */
export type Contact = Readonly<{
  /** 電話番号 */
  tel: string

  /** 続柄・関係 */
  relationship: ContactRelationship

  /** 名前 */
  name: string
}>
