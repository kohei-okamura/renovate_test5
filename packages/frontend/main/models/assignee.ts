/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */

import { StaffId } from '~/models/staff'

/**
 * 予実担当スタッフ.
 */
export type Assignee = Readonly<{
  /** スタッフID */
  staffId: StaffId | undefined

  /** 未定フラグ */
  isUndecided: boolean

  /** 研修フラグ */
  isTraining: boolean
}>
