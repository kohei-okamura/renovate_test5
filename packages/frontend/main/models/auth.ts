/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Permission } from '@zinger/enums/lib/permission'
import { Staff } from '~/models/staff'

/**
 * 認証・認可情報.
 */
export type Auth = Readonly<{
  /** システム管理者フラグ */
  isSystemAdmin: boolean

  /** 権限 */
  permissions: Permission[]

  /** スタッフ */
  staff: Staff
}>
