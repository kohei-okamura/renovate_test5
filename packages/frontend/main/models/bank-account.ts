/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { BankAccountType } from '@zinger/enums/lib/bank-account-type'
import { DateLike } from '~/models/date'

/**
 * 銀行口座 ID.
 */
export type BankAccountId = number

/**
 * 銀行口座.
 */
export type BankAccount = Readonly<{
  /** 銀行口座 ID */
  id: BankAccountId

  /** 銀行名 */
  bankName: string

  /** 銀行コード */
  bankCode: string

  /** 支店名 */
  bankBranchName: string

  /** 支店コード */
  bankBranchCode: string

  /** 種別 */
  bankAccountType: BankAccountType

  /** 口座番号 */
  bankAccountNumber: string

  /** 名義 */
  bankAccountHolder: string

  /** 登録日時 */
  createdAt: DateLike

  /** 更新日時 */
  updatedAt: DateLike
}>
