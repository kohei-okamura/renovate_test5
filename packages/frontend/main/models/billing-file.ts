/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { DwsBillingFile } from '~/models/dws-billing-file'
import { LtcsBillingFile } from '~/models/ltcs-billing-file'

export type BillingFile = DwsBillingFile | LtcsBillingFile
