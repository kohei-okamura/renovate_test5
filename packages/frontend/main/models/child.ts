/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { DateLike } from '~/models/date'
import { StructuredName } from '~/models/structured-name'

/**
 * 児童.
 */
export type Child = Readonly<{
  /** 氏名 */
  name: StructuredName

  /** 生年月日 */
  birthday: DateLike
}>
