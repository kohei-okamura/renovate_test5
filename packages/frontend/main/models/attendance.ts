/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { ServiceOption } from '@zinger/enums/lib/service-option'
import { Task } from '@zinger/enums/lib/task'
import { Assignee } from '~/models/assignee'
import { ContractId } from '~/models/contract'
import { DateLike } from '~/models/date'
import { Duration } from '~/models/duration'
import { OfficeId } from '~/models/office'
import { Schedule } from '~/models/schedule'
import { ShiftId } from '~/models/shift'
import { StaffId } from '~/models/staff'
import { UserId } from '~/models/user'

/**
 * 勤務実績 ID.
 */
export type AttendanceId = number

/**
 * 勤務実績.
 */
export type Attendance = Readonly<{
  /** 勤務実績 ID */
  id: AttendanceId

  /** 勤務シフト ID */
  shiftId: ShiftId | undefined

  /** 契約 ID */
  contractId: ContractId | undefined

  /** 事業所 ID */
  officeId: OfficeId

  /** 利用者 ID */
  userId: UserId | undefined

  /** 管理スタッフ ID */
  assignerId: StaffId

  /** 予実区分 */
  task: Task

  /** サービスコード */
  serviceCode: string

  /** 頭数 */
  headcount: number

  /** 担当スタッフ */
  assignees: Assignee[]

  /** スケジュール */
  schedule: Schedule

  /** 予実所要時間 */
  durations: Duration[]

  /** オプション */
  options: ServiceOption[]

  /** 備考 */
  note: string

  /** 確定フラグ */
  isConfirmed: boolean

  /** キャンセルフラグ */
  isCanceled: boolean

  /** キャンセル理由 */
  reason: string | undefined

  /** 登録日時 */
  createdAt: DateLike

  /** 更新日時 */
  updatedAt: DateLike
}>
