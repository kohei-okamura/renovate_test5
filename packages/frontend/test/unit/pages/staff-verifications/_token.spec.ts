/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Wrapper } from '@vue/test-utils'
import Vue from 'vue'
import { HttpStatusCode } from '~/models/http-status-code'
import StaffVerificationPage from '~/pages/staff-verifications/_token.vue'
import { createAxiosError } from '~~/test/helpers/create-axios-error'
import { createMockedApi } from '~~/test/helpers/create-mocked-api'
import { createMockedRoute } from '~~/test/helpers/create-mocked-route'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

describe('pages/staff-verifications/_token.vue', () => {
  const { mount } = setupComponentTest()
  const token = 'x'.repeat(60)
  const params = { token }
  const $route = createMockedRoute({ params })
  const $api = createMockedApi('staffs')
  const mocks = {
    $api,
    $route
  }
  let wrapper: Wrapper<Vue & any>

  async function mountComponent () {
    wrapper = mount(StaffVerificationPage, {
      mocks
    })
    await wrapper.vm.$nextTick()
  }

  function unmountComponent () {
    wrapper.destroy()
  }

  beforeEach(() => {
    jest.spyOn($api.staffs, 'verify').mockResolvedValue(undefined)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should display message when server responses 2xx', async () => {
    await mountComponent()
    expect(wrapper).toMatchSnapshot()
    unmountComponent()
  })

  it('should display message when server responses 403 Forbidden', async () => {
    jest.spyOn($api.staffs, 'verify').mockRejectedValue(createAxiosError(HttpStatusCode.Forbidden))
    await mountComponent()
    expect(wrapper).toMatchSnapshot()
    unmountComponent()
  })

  describe('validate', () => {
    const context: { params: Dictionary } = {
      params: { ...params }
    }

    beforeAll(async () => {
      await mountComponent()
    })

    afterAll(() => {
      unmountComponent()
    })

    it('should success when valid token given', () => {
      const result = wrapper.vm.$options.validate!(context)
      expect(result).toBeTrue()
    })

    it('should success when valid token given', () => {
      context.params.token = 'x'.repeat(59)
      expect(wrapper.vm.$options.validate!(context)).toBeFalse()

      context.params.token = 'x'.repeat(61)
      expect(wrapper.vm.$options.validate!(context)).toBeFalse()

      context.params.token = '-'.repeat(60)
      expect(wrapper.vm.$options.validate!(context)).toBeFalse()
    })
  })
})
