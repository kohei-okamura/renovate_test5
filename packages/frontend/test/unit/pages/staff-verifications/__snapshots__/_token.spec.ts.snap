// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/staff-verifications/_token.vue should display message when server responses 2xx 1`] = `
<v-container-stub
  fill-height=""
  tag="div"
>
  <v-row-stub
    aligncontent="center"
    justify="center"
    tag="div"
  >
    <v-col-stub
      cols="12"
      lg="4"
      md="5"
      sm="8"
      tag="div"
      xl="3"
    >
      <v-card-stub
        loaderheight="4"
        tag="div"
      >
        <z-card-titlebar-stub
          color="primary"
        >
          メールアドレス確認
        </z-card-titlebar-stub>
        <v-card-text-stub
          tag="div"
        >
          メールアドレスの確認が完了しました。
        </v-card-text-stub>
        <v-card-actions-stub
          tag="div"
        >
          <a
            block=""
            class="v-btn-stub"
            color="primary"
            nuxt=""
            text=""
            to="/"
          >
            ログイン画面へ
          </a>
        </v-card-actions-stub>
      </v-card-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;

exports[`pages/staff-verifications/_token.vue should display message when server responses 403 Forbidden 1`] = `
<v-container-stub
  fill-height=""
  tag="div"
>
  <v-row-stub
    aligncontent="center"
    justify="center"
    tag="div"
  >
    <v-col-stub
      cols="12"
      lg="4"
      md="5"
      sm="8"
      tag="div"
      xl="3"
    >
      <v-card-stub
        loaderheight="4"
        tag="div"
      >
        <z-card-titlebar-stub
          color="primary"
        >
          URL の有効期限が切れています
        </z-card-titlebar-stub>
        <v-card-text-stub
          tag="div"
        >
          お手数ですが、もう一度最初からご登録ください。
        </v-card-text-stub>
        <v-card-actions-stub
          tag="div"
        >
          <a
            block=""
            class="v-btn-stub"
            color="primary"
            nuxt=""
            text=""
            to="/staffs/new"
          >
            新規登録
          </a>
        </v-card-actions-stub>
      </v-card-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
